src/commonMain/kotlin/fr/acinq/lightning/blockchain/mempool/MempoolSpaceClient.kt

```diff
@@ -0,0 +1,147 @@
+package fr.acinq.lightning.blockchain.mempool
+
+import fr.acinq.bitcoin.Transaction
+import fr.acinq.bitcoin.TxId
+import fr.acinq.lightning.blockchain.Feerates
+import fr.acinq.lightning.blockchain.IClient
+import fr.acinq.lightning.blockchain.fee.FeeratePerByte
+import fr.acinq.lightning.logging.LoggerFactory
+import fr.acinq.lightning.logging.debug
+import fr.acinq.lightning.logging.warning
+import fr.acinq.lightning.utils.sat
+import io.ktor.client.*
+import io.ktor.client.call.*
+import io.ktor.client.plugins.*
+import io.ktor.client.plugins.contentnegotiation.*
+import io.ktor.client.request.*
+import io.ktor.client.statement.*
+import io.ktor.http.*
+import io.ktor.serialization.kotlinx.json.*
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json
+
+class MempoolSpaceClient(val mempoolUrl: Url, loggerFactory: LoggerFactory) : IClient {
+
+    private val logger = loggerFactory.newLogger(this::class)
+
+    @OptIn(ExperimentalSerializationApi::class)
+    val client = HttpClient {
+        install(ContentNegotiation) {
+            json(json = Json {
+                prettyPrint = true
+                isLenient = true
+                explicitNulls = false // convert absent fields to null
+                ignoreUnknownKeys = true
+            })
+        }
+        install(DefaultRequest) {
+            url {
+                takeFrom(mempoolUrl)
+            }
+        }
+    }
+
+    suspend fun publish(tx: Transaction) {
+        val res = client.post("api/tx") {
+            contentType(ContentType.Text.Plain)
+            setBody(tx.toString())
+        }
+        if (!res.status.isSuccess()) {
+            logger.warning { "tx publish failed for txid=${tx.txid}: ${res.bodyAsText()}" }
+        }
+    }
+
+    suspend fun getTransaction(txId: TxId): Transaction? {
+        return kotlin.runCatching {
+            val res = client.get("api/tx/$txId/hex")
+            if (res.status.isSuccess()) {
+                Transaction.read(res.bodyAsText())
+            } else null
+        }.onFailure { logger.warning(it) { "error in getTransaction " } }
+            .getOrNull()
+    }
+
+    suspend fun getTransactionMerkleProof(txId: TxId): MempoolSpaceTransactionMerkleProofResponse? {
+        return kotlin.runCatching {
+            val res = client.get("api/tx/$txId/merkle-proof")
+            if (res.status.isSuccess()) {
+                val txStatus: MempoolSpaceTransactionMerkleProofResponse = res.body()
+                txStatus
+            } else null
+        }.onFailure { logger.warning(it) { "error in getTransactionMerkleProof " } }
+            .getOrNull()
+    }
+
+    suspend fun getOutspend(txId: TxId, outputIndex: Int): Transaction? {
```

Comment:
What does `Outspend` mean? Some documentation on the function in this file would be useful.

Reply:
Done in 0724e56a.